core/common/src/main/java/alluxio/worker/block/io/IOFunction.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.block.io;
+
+import java.io.IOException;
+
+/**
+ * A callable for IO functions.
+ */
+public interface IOFunction {
+  /**
+   * Compute with some io functions.
+   * @throws IOException
+   */
+  void call() throws IOException;
```

Comment:
it looks throwing IOE here makes the `catch` clause tricky.
Actually we are not really handling the IOE when abort fails, e.g. in following code like

```java
catch (Exception e) {
        reader.abort();
        if (e instanceof IOException) {
          throw (IOException) e;
        }
        throw new RuntimeException(e);
      }
}
```

For simplicity, how about just use `Runnable` interface (without being able to throw IOE)?

Reply:
I had tried Runnable, the abort and commit method will wrap the IOE into RuntimeException, which might break the current exception handling chain. I'm not sure how the callers of the method  I modified are handling the IOE.  To be safe, I still prefer to keep current logic, if it's an IOE, I still throw an IOE.